pygmt/clib/session.py

```diff
@@ -2311,3 +2312,29 @@ def extract_region(self) -> np.ndarray:
         if status != 0:
             raise GMTCLibError("Failed to extract region from current figure.")
         return region
+
+
+def raster_kind(raster: str):
```

Comment:
Add type hint, and make this into a private function.

```suggestion
def _raster_kind(raster: str) -> Literal["grid", "image"] | None:
```